ndcube/mixins/ndslicing.py

```diff
@@ -19,10 +19,25 @@ def __getitem__(self, item):
         if item is None or (isinstance(item, tuple) and None in item):
             raise IndexError("None indices not supported")
 
+        # If cube has a sliceable metadata, remove it and handle it separately.
+        # This is to prevent the shapes of the data and metadata getting out of
+        # sync part way through the slicing process.
+        meta_is_sliceable = False
+        if hasattr(self.meta, "__ndcube_can_slice__") and self.meta.__ndcube_can_slice__:
+            meta_is_sliceable = True
+            meta = self.meta
+            self.meta = None
```

Comment:
Does this not mean that the original cube you are slicing now has no metadata after slicing it?

Reply:
Yes.  It's fixed now.